### ISSUE TITLE
cannot get revisions when creating and updating a workload

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#16408]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1.  run  rancher server v2.1.2-rc3 
2.  add a ec2 cluster with cloud provider from AWS, 5 nodes(including 3 workers)
3.  deploy a workload, after the workload is active, go to **view in API**
<img width="1718" alt="1" src="https://user-images.githubusercontent.com/6218999/47871728-0f9ed580-ddca-11e8-9acb-bd887857c4bb.png">

4.  go to the link to revisions
<img width="821" alt="2" src="https://user-images.githubusercontent.com/6218999/47871735-13caf300-ddca-11e8-92cb-c8d022b64bc4.png">

5. check the data section for revisions
<img width="737" alt="3" src="https://user-images.githubusercontent.com/6218999/47871748-19283d80-ddca-11e8-89cf-1c2ebde7b257.png">

**Result:**
The data section is empty. There should be at one entry which is the workload itself. 

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.1.2-c3
- Installation option (single install/HA):
single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
Infrastructure Provide
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
could, m2.medium
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T17:53:03Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker (use `docker info`):

```
Containers: 45
 Running: 25
 Paused: 0
 Stopped: 20
Images: 17
Server Version: 17.03.2-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 221
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-1070-aws
Operating System: Ubuntu 16.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858 GiB
Name: jiaqi-v2-1-2-rc3-3
ID: LIUY:J5D7:PN4C:O5FA:BIYR:W2WC:TOI6:MJCZ:4MZV:7U6I:CJCF:PSLG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
 provider=amazonec2
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by kinarashah** [comment]
While testing, we also need to revalidate https://github.com/rancher/rancher/issues/15804

**Comment by jiaqiluo** [comment]
## Reproduction
### Steps
1. run rancher server v2.1.2-rc3
1. add a ec2 cluster with cloud provider from AWS, 5 nodes(including 3 workers)
1. deploy a workload, after the workload is active, go to view in API
1. go to the link to revisions
1. check the data section for revisions
### Result:
The same bug is reproduced: 
The data section is empty. There should be at one entry which is the workload itself.

## Validation
1. run rancher server v2.1.2-rc4
1. add a ec2 cluster with cloud provider from AWS, 5 nodes(including 3 workers)
1. deploy a workload, after the workload is active, go to view in API
1. go to the link to revisions
1. check the data section for revisions
  <img width="790" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/48025410-8f95aa00-e100-11e8-99c6-17c4fa7b7ca6.png">

1. update the workload by adding new environment variable
6. go to view in API -> go to the link to revisions -> check the data section for revisions.
<img width="652" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/48025682-57429b80-e101-11e8-925c-ed2e9cc50ed6.png">

### Result: 
The data section contains the correct number of revisions.

## Conclusion
The bug is fixed in v2.1.2-rc4.

**Comment by jiaqiluo** [comment]
> While testing, we also need to revalidate #15804

## Revalidation #15804 on Rnacher:v2.1.2-rc4
**Type/provider of hosts**:  Imported GKE v1.10.7-gke.6 cluster
**Steps** : 
1. Create a GKE cluster through gcloud.
1. Import GKE cluster.
1. Launch Applications(nginx, prometheus) through the GCP Application Marketplace.

**Result**: 
Pods show in `Group By Namespace/Workload`, `Group by Workload`,  `flat list` and `Group by Node`. 
<img width="1832" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/48034463-beba1480-e11c-11e8-86e9-8defd2b028b0.png">
<img width="1801" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/48034468-c2e63200-e11c-11e8-84f5-7151b161769f.png">
